components/admin_console/system_users/system_users.jsx

```diff
@@ -135,15 +139,19 @@ export default class SystemUsers extends React.Component {
         this.props.actions.setSystemUsersSearch(term, this.props.teamId);
     }
 
-    nextPage(page) {
+    nextPage = (page) => {
         // Paging isn't supported while searching
 
         if (this.props.teamId === SearchUserTeamFilter.ALL_USERS) {
             loadProfiles(page + 1, USERS_PER_PAGE, this.loadComplete);
         } else if (this.props.teamId === SearchUserTeamFilter.NO_TEAM) {
             loadProfilesWithoutTeam(page + 1, USERS_PER_PAGE, this.loadComplete);
         } else {
-            loadProfilesAndTeamMembers(page + 1, USERS_PER_PAGE, this.props.teamId, this.loadComplete);
+            this.props.actions.loadProfilesAndTeamMembers(page + 1, USERS_PER_PAGE, this.props.teamId).then(({data}) => {
```

Comment:
Maybe, if this not propage the async to other functions, makes sense to keep this function consistent with the `loadDataForTeam` function. I mean, set the function as `async` and use `await` instead of `.then`.

Reply:
Sure, changing since I'm 0/5 with that and yours make sense 💯